### PR TITLE
Fix Request Connection Issues

### PR DESCRIPTION
**Pull request by msgerbush**
Fixed a couple of issues with the API connection classes:
- Basic auth string formatter required a second argument
- Connection object was not being assigned on new ApiConnections
- The base address of a url and the content-type were not being sent
- User-agent string was not a valid format because missing .NET version
- Removed unused(?) clone method that was causing an error
- Manage System.Net.Http through NuGet because I was running into a bug with an older version: https://github.com/dotnet/corefx/issues/9846

### CODE REVIEW

**Comment by virtualdom** [comment]
Hi @msgerbush, thanks for putting up this PR with fixes for the API connection classes. We'll take a look at it soon.

**Comment by msgerbush** [comment]
Hi, any updates on either of these pull requests?  Is this library actively being worked on?

**Comment by jwslob** [comment]
Hi @msgerbush , we have it as a task in the sprint starting next week. Thanks for your patience.

**Comment by msgerbush** [comment]
Great, thank you! Is there a better channel to ask questions?  I don't want to spam you with new issues or pull requests and I don't want to overlap on planned new work or features.

**Comment by ami** [comment]
@msgerbush Of course! Send us a support ticket [here](https://lob.com/support#contact) for general inquires/questions. 😄

**Comment by msgerbush** [comment]
Oh, not so much general inquiries but questions about this .NET library.  I realize it's in alpha, but it would be helpful to know if certain features are already planned, or if I would need to add them on my own.  And if I need to add them, I would prefer to do it with feedback from the Lob team, so I don't waste time on features that won't get accepted into the project.

**Comment by ami** [comment]
That's the right avenue for that as well! Just pick "Technical Support".

File: Lob/Authentication/BasicAuthenticator.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public void Authenticate(IRequest request, Credentials credentials)
                 CultureInfo.InvariantCulture,
                 "Basic {0}",
                 Convert.ToBase64String(Encoding.UTF8.GetBytes(
-                    string.Format(CultureInfo.InvariantCulture, "{0}:{1}", credentials.ApiKey))));
+                    string.Format(CultureInfo.InvariantCulture, "{0}:{1}", credentials.ApiKey, null))));
```
Comment (jwslob) [inline Lob/Authentication/BasicAuthenticator.cs]: nit, why not just `"{0}:"`

File: Lob/Http/Connection.cs [hunk 2]
```diff
@@ -123,7 +125,7 @@ public Credentials Credentials
 
         static string FormatUserAgent(ProductHeaderValue productInformation)
         {
-            return string.Format(CultureInfo.InvariantCulture, "Lob/v1 C# .NET/", productInformation);
+            return string.Format(CultureInfo.InvariantCulture, "Lob/v1 C# .NET/4.6", productInformation);
```
Comment (jwslob) [inline Lob/Http/Connection.cs]: `productInformation` isn't used. it is more clear if we just remove it from the string format for now.

File: Lob/Authentication/BasicAuthenticator.cs (hunk 1 above)
Comment (msgerbush) [reply Lob/Authentication/BasicAuthenticator.cs]: Yeah, wasn't sure if that was a stylistic choice or something

**Pull request merged**